ksql-rest-app/src/main/java/io/confluent/ksql/api/server/ServerVerticle.java

```diff
@@ -49,26 +46,25 @@
   private final Endpoints endpoints;
   private final HttpServerOptions httpServerOptions;
   private final Server server;
-  private final boolean proxyEnabled;
+  private final ProxyHandler proxyHandler;
   private ConnectionQueryManager connectionQueryManager;
   private volatile HttpServer httpServer;
 
-  private HttpClient proxyClient;
-  private SocketAddress proxyTarget;
-
   public ServerVerticle(final Endpoints endpoints, final HttpServerOptions httpServerOptions,
       final Server server, final boolean proxyEnabled) {
     this.endpoints = Objects.requireNonNull(endpoints);
     this.httpServerOptions = Objects.requireNonNull(httpServerOptions);
     this.server = Objects.requireNonNull(server);
-    this.proxyEnabled = proxyEnabled;
+    if (proxyEnabled) {
+      this.proxyHandler = new ProxyHandler(server);
+    } else {
+      this.proxyHandler = null;
+    }
   }
 
   @Override
   public void start(final Promise<Void> startPromise) {
-    this.proxyClient = proxyEnabled ? vertx
```

Comment:
Love the refactor of proxy client and target handling from this file into `ProxyHandler` -- much cleaner! Thanks :)